### ISSUE TITLE
input-container: ng-messages are no longer hidden on form reset as of 1.1.1

### ISSUE THREAD

**Issue by tom94zoe** [angular/material#9543]
**Actual Behavior**:
-  I tried to hide ng-message with the following code, since Angular Material 1.1.1 the message wasn't hidden on reset. 

``` html
<md-input-container flex style="margin-top: 0;margin-bottom: 5px">
        <label translate>maintenanceMessage.description</label>
        <input md-maxlength="40" required name="description" md-no-asterisk
               ng-model="maintenanceMessageCtrl.newMaintenanceMessage.description" flex>
        <div ng-messages="maintenanceMessageForm.description.$error" ng-show="maintenanceMessageForm.description.$touched">
          <div ng-message="required" translate>maintenanceMessage.description.requiredMessage</div>
          <div ng-message="md-maxlength" translate>maintenanceMessage.description.maxLengthMessage</div>
        </div>
</md-input-container>
```

``` js
ctrl.resetFormAndHideDialog = function () {
      ctrl.newMaintenanceMessage = {};
      $scope.maintenanceMessageForm.$setUntouched();
      $scope.maintenanceMessageForm.$setPristine();
     };
```

**Angular Versions**: *
-  `Angular Version:1.5.8
-  `Angular Material Version:`1.1.1

![angular-material-ng-message](https://cloud.githubusercontent.com/assets/17591511/18313018/cd280370-750c-11e6-964d-939f1714fdb6.png)

**Comment by mjanthrax** [comment]
yea me too, i'm experiencing that kind of issue. so i decided to go back to Angular Material Version 1.1.0rc5

**Comment by mjanthrax** [comment]
hope someone can help us here. cause i really need the new browser color feature of new update in angular material

**Comment by tom94zoe** [comment]
That would be fine, but I think this bug will be fixed with 1.1.3. So we have to wait some weeks.

**Comment by ThomasBurleson** [comment]
@tom94zoe - can you provide a CodePen for use to verify fixes?

**Comment by rhinoceraptor** [comment]
@ThomasBurleson Here is a codepen I made showing the bug:

[https://codepen.io/rhinocerapt0r/pen/ORBQJZ?editors=1011](https://codepen.io/rhinocerapt0r/pen/ORBQJZ?editors=1011)

**Comment by clshortfuse** [comment]
@rhinoceraptor Thanks! This does confirm my suspicion that this is related to ngAnimate. I believe I had commented before on a similar issue where I mentioned we have solution built in an internal PR. The code will need to be cherrypicked.

**Comment by rhinoceraptor** [comment]
If you change the version of the Angular material CSS and JS to 1.0.9, you can see it working properly.

**Comment by rhinoceraptor** [comment]
I think [this](https://github.com/angular/material/issues/1903) is a duplicate issue. I couldn't downgrade my version of angular material, so I used this CSS workaround:

```
.ng-untouched ~ .md-input-messages-animation {
  opacity: 0;
}
```

**Comment by samithaf** [comment]
@rhinoceraptor the fix is working but there is a small UI defect and can be fixed by adding following.

```
 .ng-untouched ~ .md-input-messages-animation {
    height: 0;
    opacity: 0;
  }
```

**Comment by topherfangio** [comment]
I'm removing the `has: Pull Request` label because that PR does not properly fix all of the issues. We will try to provide a different fix for this.

**Comment by Splaktar** [comment]
This is fixed in 1.1.10 as seen in this [updated CodePen](https://codepen.io/Splaktar/pen/qypOKd?editors=1011).
